library/extractor/src/main/java/com/google/android/exoplayer2/extractor/flv/ScriptTagPayloadReader.java

```diff
@@ -89,6 +89,9 @@ protected boolean parsePayload(ParsableByteArray data, long timeUs) {
       // We're only interested in metadata.
       return false;
     }
+    if (data.getData().length <= data.getPosition()) {
```

Comment:
Can you use "data.bytesLeft() == 0" instead and log a warning?

Note: I can't test myself because the file you provided doesn't work for me (either in ExoPlayer or in FFMPEG).

Reply:
I am sorry you can't test yourself. Maybe you've heard of GFW in China. Anyway, I've test myself and "data.bytesLeft() == 0" is fine. I think it's a better way, too. So I changed it.